add doc for new-collation-enable in br (#8031)
pingcap/docs: br/backup-and-restore-tool.md

| Use BR nightly to back up TiDB nightly | ✅ | ✅ | ❌ (If a table with the primary key of the non-integer clustered index type is restored to a TiDB v4.0 cluster, BR will cause data error without warning.)  |
| Use BR v5.0 to back up TiDB v5.0 | ✅ | ✅ | ❌  (If a table with the primary key of the non-integer clustered index type is restored to a TiDB v4.0 cluster, BR will cause data error without warning.)
| Use BR v4.0 to back up TiDB v4.0 | ✅ | ✅ | ✅ (If TiKV >= v4.0.0-rc.1, and if BR contains the [#233](https://github.com/pingcap/br/pull/233) bug fix and TiKV does not contain the [#7241](https://github.com/tikv/tikv/pull/7241) bug fix, BR will cause the TiKV node to restart.) |
| Use BR nightly or v5.0 to back up TiDB v4.0 | ❌ (If the TiDB version is earlier than v4.0.9, the [#609](https://github.com/pingcap/br/issues/609) issue might occur.) | ❌ (If the TiDB version is earlier than v4.0.9, the [#609](https://github.com/pingcap/br/issues/609) issue might occur.) | ❌ (If the TiDB version is earlier than v4.0.9, the [#609](https://github.com/pingcap/br/issues/609) issue might occur.) |

#### Check for the `new_collations_enabled_on_first_bootstrap` variable

> Since TiDB v6.0.0, the default value of [`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) has changed from `false` to `true`. When the value `new_collations_enabled_on_first_bootstrap` is consistent between the upstream and downstream clusters, BR safely restores the backup data of the upstream cluster to the downstream cluster.

Reviewer: ```suggestion
Since TiDB v6.0.0, the default value of [`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) changes to `true` from `false`. When the value of `new_collations_enabled_on_first_bootstrap` is consistent between the upstream and downstream clusters, BR safely restores the data backed up in the upstream cluster to the downstream cluster.
```

> Since v6.0.0, BR backs up the `new_collations_enabled_on_first_bootstrap` configuration of the upstream cluster and then checks whether this configuration is consistent between the upstream and downstream clusters. If the value of `new_collations_enabled_on_first_bootstrap` is inconsistent between the upstream and downstream clusters, BR does not perform data restore and reports an error.

Reviewer: ```suggestion
Since v6.0.0, BR backs up the `new_collations_enabled_on_first_bootstrap` configuration of the upstream cluster and then checks whether the value of this configuration is consistent between the upstream and downstream clusters. If the value is inconsistent between the upstream and downstream clusters, BR does not perform the data restore and reports an error.
```

> If you need to restore data in a TiDB cluster of an earlier version to TiDB v6.0.0 or a later version, you need manually to check whether the value of `new_collations_enabled_on_first_bootstrap` is consistent between the upstream and downstream clusters:

Reviewer: ```suggestion
Suppose that you have backed up the data in a TiDB cluster of an earlier version of v6.0.0, and you want to restore this data to a TiDB cluster of v6.0.0 or later versions. In this situation, you need manually to check whether the value of `new_collations_enabled_on_first_bootstrap` is consistent between the upstream and downstream clusters:
```

- If the value is consistent, you can add `--check-requirements=false` to the restore command to skip this configuration check.

> - If the value is inconsistent, and you forcibly perform the restore, BR reports a [data validation error](/br/backup-and-restore-tool.md#usage-restrictions).

Reviewer 2: ```suggestion
- If not, and you forcibly perform the restore, BR could report a [data validation error](/br/backup-and-restore-tool.md#usage-restrictions).
```

> - If the `new_collations_enabled_on_first_bootstrap` configuration value is consistent, you can add `--check-requirements=false` to the restore command to skip this configuration check.
> - If the `new_collations_enabled_on_first_bootstrap` configuration value is inconsistent and you forcibly perform the restore, BR will report a [data validation error](/br/backup-and-restore-tool.md#usage-restrictions).

Reviewer: ```suggestion
- If the value is consistent, you can add `--check-requirements=false` to the restore command to skip this configuration check.
- If the value is inconsistent, and you forcibly perform the restore, BR will report a [data validation error](/br/backup-and-restore-tool.md#usage-restrictions).
```
